File: django/jge/db_automation/blog_post/repository/blog_post_repository.py
from abc import ABC, abstractmethod

from blog_post.entity.blog_post import BlogPost


class BlogPostRepository(ABC):

    @abstractmethod
    def list(self, page, perPage):
        pass

    @abstractmethod
    def save(self, blog_post: BlogPost) -> BlogPost:
        pass

    @abstractmethod
    def findById(self, boardId):
        pass

File: flutter/jge/first/lib/blog_post/infrastructures/repository/blog_post_repository.dart
import '../../domain/entity/blog_post.dart';
import '../../domain/usecases/list/response/blog_post_list_response.dart';

abstract class BlogPostRepository {
  Future<BlogPostListResponse> requestList(int page, int perPage);
  Future<String> uploadBlogPost(String title, String compressedHtmlContent, String userToken);
  Future<BlogPost> create(String title, String content, String userToken);
  Future<BlogPost?> readBlogPost(int id);
  Future<String> uploadBlogPostImage(String imageContent, String userToken);
  Future<BlogPost?> updateBlogPost(
      int blogPostId, String title, String content, String userToken);
  Future<void> deleteBlogPost(int id, String userToken);
}